app.py

```diff
@@ -1,134 +1,172 @@
-import os
-from flask import Flask, render_template, jsonify, request
+from flask import Flask, render_template, jsonify, request, Response
 import requests
-from urllib.parse import urlparse, quote_plus
+from urllib.parse import quote_plus
```

Comment:
The addition of type annotations enhances code clarity and helps in catching potential type-related errors. This is a welcome improvement and can be beneficial for maintenance and refactoring.